### PR TITLE
Add mentor and mentee search

### PR DESCRIPTION
**Pull request by lakshay-angrish**

### CODE REVIEW

File: models/mentee.js [hunk 1]
```diff
@@ -0,0 +1,103 @@
+var mongoose = require("mongoose");
+var passportLocalMongoose = require("passport-local-mongoose");
```
Comment (deepakpixel) [inline models/mentee.js:2]: no need of passport-local-mongoose dependency

File: models/mentee.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    },
+    mentorAssign: {
+      type: String,
+      default: "none",
+    },
+  },
+  { timestamps: true }
+);
+
+MenteeSchema.plugin(passportLocalMongoose, {
```
Comment (deepakpixel) [inline models/mentee.js:99]: that also goes away with passporlocalmongoose,
similarly for mentor profile

File: models/mentor.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+var mongoose = require("mongoose");
+var passportLocalMongoose = require("passport-local-mongoose");
+var MentorSchema = mongoose.Schema({
+  name: String,
+  password: String,
+  username: String,
+  phone: String,
+  whatsapp: String,
```
Comment (deepakpixel) [inline models/mentor.js:8]: add a college field here

Comment (lakshay-angrish) [reply models/mentor.js:8]: done

File: routes/adminRouter.js [hunk 4]
```diff
@@ -37,4 +39,122 @@ router.get("/msg91-balance", checkAuth, async (req, res, next) => {
...
+function escapeRegex(text) {
+  return text.replace(/[-[\]{}()*+?.,\\^$|#\s]/g, "\\$&");
+}
+
+router.post("/menteeSearch", checkAuth, async (req, res, next) => {
+  try {
+    if (
+      !req.body.searchType ||
+      !req.body.searchQuery ||
+      !req.body.page ||
```
Comment (deepakpixel) [inline routes/adminRouter.js:51]: crashes when perpage and page are not numbers.

File: models/mentor.js (hunk 3 above)
Comment (lakshay-angrish) [reply models/mentor.js:7]: done

File: models/mentee.js (hunk 2 above)
Comment (lakshay-angrish) [reply models/mentee.js]: done

File: models/mentee.js (hunk 1 above)
Comment (lakshay-angrish) [reply models/mentee.js]: done

File: routes/adminRouter.js (hunk 4 above)
Comment (lakshay-angrish) [reply routes/adminRouter.js:51]: done

**Review by deepakpixel — CHANGES_REQUESTED** [review]
'return' the responses, once caught into: Cannot set headers after they are sent to the client. Everything else seems to be fine

**Review by deepakpixel — APPROVED** [review]

**Review by deepakpixel — CHANGES_REQUESTED** [review]
.count() is depricated use .countDocuments()
https://mongoosejs.com/docs/deprecations.html

**Review by deepakpixel — APPROVED** [review]

**Pull request merged**